agent/grpc-external/services/resource/watch.go

```diff
@@ -33,6 +57,16 @@ func (s *Server) WatchList(req *pbresource.WatchListRequest, stream pbresource.R
 			continue
 		}
 
+		// filter out items that don't pass read ACLs
+		if err = reg.ACLs.Read(authz, event.Resource.Id); err != nil {
+			switch {
+			case acl.IsErrPermissionDenied(err):
+				continue
+			default:
+				return fmt.Errorf("authorizing read: %w", err)
+			}
+		}
+
```

Comment:
nit: you could reduce the level of nesting here by combining the `if` and `switch`.

```suggestion
		err = reg.ACLs.Read(authz, event.Resource.Id)
		switch {
		case acl.IsErrPermissionDenied(err):
			continue
		case err != nil:
			return fmt.Errorf("authorizing read: %w", err)
		}
```

Reply:
👍 Fixed